From 553aeaa383162c0f543f971119c68ee621a511d1 Mon Sep 17 00:00:00 2001
From: Huiquan Zhong <huiquan.zhong@intel.com>
Date: Mon, 28 Sep 2015 17:58:31 +0800
Subject: [PATCH] dma: dw: remove ->shutdown() hook

Don't touch DMA register as LPSS maybe power gated in ->shutdown() hook.

Change-Id: If4b59798d16929b39a16dd7023b7a3e199a90b3b
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-5122
Signed-off-by: Huiquan Zhong <huiquan.zhong@intel.com>
---
 drivers/dma/dw/core.c     | 10 ----------
 drivers/dma/dw/platform.c |  8 --------
 2 files changed, 18 deletions(-)

diff --git a/drivers/dma/dw/core.c b/drivers/dma/dw/core.c
index 1ece4c5..66e5d34 100644
--- a/drivers/dma/dw/core.c
+++ b/drivers/dma/dw/core.c
@@ -1669,16 +1669,6 @@ int dw_dma_remove(struct dw_dma_chip *chip)
 }
 EXPORT_SYMBOL_GPL(dw_dma_remove);
 
-void dw_dma_shutdown(struct dw_dma_chip *chip)
-{
-	struct dw_dma *dw = chip->dw;
-
-	dw_dma_off(dw);
-	if (dw->clk)
-		clk_disable_unprepare(dw->clk);
-}
-EXPORT_SYMBOL_GPL(dw_dma_shutdown);
-
 #ifdef CONFIG_PM_SLEEP
 
 int dw_dma_suspend(struct dw_dma_chip *chip)
diff --git a/drivers/dma/dw/platform.c b/drivers/dma/dw/platform.c
index 5389d09..618a3b6 100644
--- a/drivers/dma/dw/platform.c
+++ b/drivers/dma/dw/platform.c
@@ -236,13 +236,6 @@ static int dw_remove(struct platform_device *pdev)
 	return dw_dma_remove(chip);
 }
 
-static void dw_shutdown(struct platform_device *pdev)
-{
-	struct dw_dma_chip *chip = platform_get_drvdata(pdev);
-
-	dw_dma_shutdown(chip);
-}
-
 #ifdef CONFIG_OF
 static const struct of_device_id dw_dma_of_id_table[] = {
 	{ .compatible = "snps,dma-spear1340" },
@@ -294,7 +287,6 @@ static const struct dev_pm_ops dw_dev_pm_ops = {
 static struct platform_driver dw_driver = {
 	.probe		= dw_probe,
 	.remove		= dw_remove,
-	.shutdown	= dw_shutdown,
 	.driver = {
 		.name	= DRV_NAME,
 		.pm	= &dw_dev_pm_ops,
-- 
1.9.1

